src/main/java/com/thealgorithms/datastructures/graphs/WelshPowell.java

```diff
@@ -0,0 +1,73 @@
+package com.thealgorithms.datastructures.graphs;
+
+import java.util.LinkedList;
+
+/*
+ *  The Welsh-Powell algorithm is a graph coloring algorithm
+ *  used for coloring a graph with the minimum number of colors.
+ *  https://en.wikipedia.org/wiki/Graph_coloring
+ */
+public class WelshPowell {
+
+    public static class WPGraph {
+
+        private int numVer; // Number of vertices in the graph
+        public LinkedList<Integer>[] adjLists; // Adjacency lists for the graph
+
+        // Constructor to initialize the graph with a certain number of vertices
+        public WPGraph(int vertices) {
+            numVer = vertices;
+            adjLists = new LinkedList[vertices];
+
+            for (int i = 0; i < vertices; i++) {
+                adjLists[i] = new LinkedList<>();
+            }
+        }
+
+        // Method to add an edge to the graph (undirected)
+        public void addEdge(int src, int dest) {
```

Comment:
I would suggest to rename this arguments. Current names somehow suggest that you add _directed edges_.

```suggestion
        public void addEdge(int nodeA, int nodeB) {
```